smartcontactmanager/src/main/java/com/scm/smartcontactmanager/service/contact/ContactServiceImpl.java

```diff
@@ -1,8 +1,128 @@
+		}
+		Contact contact = new Contact();
+		contact.setFirstName(contactRequestDto.getFirstName());
+		contact.setMiddleName(contactRequestDto.getMiddleName());
+		contact.setLastName(contactRequestDto.getLastName());
+		contact.setContactNumber(contactRequestDto.getContactNumber());
+		contact.setPersonalEmail(contactRequestDto.getPersonalEmail());
+		contact.setWorkEmail(contactRequestDto.getWorkEmail());
+		contact.setHomeAddress(contactRequestDto.getHomeAddress());
+		contact.setWorkAddress(contactRequestDto.getWorkAddress());
+		contact.setDepartment(contactRequestDto.getDepartment());
+		contact.setWebsite(contactRequestDto.getWebsite());
+		contact.setUpdatedBy("USER");
+		return contactDao.createContact(contact);
+	}
+
+	@Override
+	public void deleteAllContact() {
+		contactDao.deleteallContact();
+	}
+
+//	@Override
+//	public Contact getallContact(ContactRequestDto contactRequestDto) {
+//		Contact contact=new Contact();
+//		contact.setFirstName(contactRequestDto.getFirstName());
+//		contact.setContactNumber(contactRequestDto.getContactNumber());
+//		contact.setPersonalEmail(contactRequestDto.getPersonalEmail());
+//		return contactDao.getallContact(contact);
+//	}
+
+	@Override
+	public void updateContactById(Contact contactDto, int id) {
+		Optional<Contact> updateContactinfo = contactDao.getcontactById(id);
+		if(!updateContactinfo.isPresent()) {
+			throw new ServiceException(HttpStatusCode.valueOf(404), "Didn Found ID: " + id);
+		}
+		Contact contact = new Contact();
+		contact.setId(id);
+		contact.setFirstName(contactDto.getFirstName());
+		contact.setMiddleName(contactDto.getMiddleName());
+		contact.setLastName(contactDto.getLastName());
+		contact.setContactNumber(contactDto.getContactNumber());
+		contact.setPersonalEmail(contactDto.getPersonalEmail());
+		contact.setWorkEmail(contactDto.getWorkEmail());
+		contact.setHomeAddress(contactDto.getHomeAddress());
+		contact.setWorkAddress(contactDto.getWorkAddress());
+		contact.setDepartment(contactDto.getDepartment());
+		contact.setWebsite(contactDto.getWebsite());
+		contact.setUpdatedBy("USER");
+		
+		 contactDao.createContact(contact);
+
+	}
+
+	@Override
+	public void deleteById(int id) {
+		Optional<Contact>deleteByid=contactDao.getcontactById(id);
+		if(!deleteByid.isPresent()) {
+			throw new ServiceException(HttpStatusCode.valueOf(409), "Id didn't Found in Data base:"+id);
+		}
+		contactDao.deleteById(id);
+
+	}
+
+	@Override
+	public Contact getcontactById(int id) {
+		Optional<Contact> contactInfo = contactDao.getcontactById(id);
+		if (!contactInfo.isPresent()) {
+			throw new ServiceException(HttpStatusCode.valueOf(404), "No information with id: " + id);
+		}
+		return contactInfo.get();
+	}
 
+	@Override
+	public List<Contact> getAllContact(PageRequest pageRequest, String name, String email) {
+		List<Contact> contactList = new ArrayList<>();
+		if (name != null && !name.isEmpty()) {
```

Comment:
Use StringUtils of apache to do null check

Reply:
Resolved.